adafruit_is31fl3731/__init__.py

```diff
@@ -108,36 +118,40 @@ def _i2c_read_reg(self, reg, result):
             return result
         return None
 
-    def _i2c_write_reg(self, reg, data):
+    def _i2c_write_reg(self, reg: int = None, data: bytes = None) -> bytes:
         # Write a contiguous block of data (bytearray) starting at the
         # specified I2C register address (register passed as argument).
         self._i2c_write_block(bytes([reg]) + data)
 
-    def _i2c_write_block(self, data):
+    def _i2c_write_block(self, data: bytes = None) -> bytes:
         # Write a buffer of data (byte array) to the specified I2C register
         # address.
         with self.i2c_device as i2c:
             i2c.write(data)
 
-    def _bank(self, bank=None):
+    def _bank(self, bank: int = None) -> int:
         if bank is None:
             result = bytearray(1)
             return self._i2c_read_reg(_BANK_ADDRESS, result)[0]
         self._i2c_write_reg(_BANK_ADDRESS, bytearray([bank]))
         return None
 
-    def _register(self, bank, register, value=None):
+    def _register(
+        self, bank: int = None, register: int = None, value: int = None
+    ) -> int:
         self._bank(bank)
         if value is None:
             result = bytearray(1)
+            print(f"Register: {result}")
             return self._i2c_read_reg(register, result)[0]
         self._i2c_write_reg(register, bytearray([value]))
         return None
 
     def _mode(self, mode=None):
+        """Function for setting _register mode"""
         return self._register(_CONFIG_BANK, _MODE_REGISTER, mode)
 
-    def _init(self, frames=None):
+    def _init(self, frames: int = 0) -> int:
```

Comment:
It iterates through integers 0-7 so I figured this one would be an int. :/  Didn't know about the Iterable type, that will complicate things.

Reply:
It can't be an `int` because then this for loop ` for frame in frames:` will raise this exception `TypeError: 'int' object is not iterable`

Sorry for the confusion of mentioning it was an int earlier I didn't look far enough down the chain of it's usage to figure out what was being done with it. The prior comment was more about the `Optional[]` than the specific type within it, but I should have looked close at the time.

I was a little unsure about what these frames were but I did find in the intro of the datasheet that this chip supports 8 frames of an animation that it can keep in memory and then playback with given delays https://lumissil.com/assets/pdf/core/IS31FL3731_DS.pdf this could be used to offload the work of the animation from the MCU into this driver. I'm assuming that is what the `frames` arguments here are dealing with.

I had a look through the rest of the code in this library and I cannot find anywhere that is actually using any value for `frames` so it's always going with the default and then using `range(8)` to end up with the list of numbers 0 thru 7.

Reply:
it gets further complicated by the fact that `autoplay` function also has a `frames` argument, but that one is actually and `int` not an `Iterable`